sound/soc/sof/debug.c

```diff
@@ -17,6 +17,103 @@
 #include "sof-priv.h"
 #include "ops.h"
 
+static ssize_t sof_dfsentry_write(struct file *file, const char __user *buffer,
+				  size_t count, loff_t *ppos)
+{
+	struct snd_sof_dfsentry *dfse = file->private_data;
+	struct snd_sof_dev *sdev = dfse->sdev;
+	struct sof_ipc_cmd_hdr hdr;
+	struct sof_ipc_reply reply;
+	size_t size;
+	char *string;
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_DEBUG)
+	ktime_t start, end;
+	s64 ipc_response_time;
+	s64 avg_response_time = 0;
+	s64 min_response_time = LONG_MAX;
+	s64 max_response_time = 0;
+	int i, err, ret;
+	unsigned long ipc_count;
+#endif
+	string = kzalloc(count, GFP_KERNEL);
+	if (!string)
+		return -ENOMEM;
+
+	size = simple_write_to_buffer(string, count, ppos, buffer, count);
+	ret = size;
+
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_DEBUG)
+	/* check if "ipc_test" debugfs entry */
+	if (strcmp(dfse->dfsentry->d_name.name, "ipc_test"))
+		return -EINVAL;
+
+	ret = pm_runtime_get_sync(sdev->dev);
+	if (ret < 0) {
+		dev_err_ratelimited(sdev->dev,
+				    "error: enum put failed to resume %d\n",
+				    ret);
+		pm_runtime_put_noidle(sdev->dev);
+		return ret;
+	}
+
+	ret = kstrtoul(string, 10, &ipc_count);
+	if (ret < 0)
+		return ret;
+
+	/* configure test IPC */
+	hdr.cmd = SOF_IPC_GLB_TEST | SOF_IPC_TEST_IPC_FLOOD;
+	hdr.size = sizeof(hdr);
+
+	/* send test IPC's */
+	for (i = 0; i < ipc_count; i++) {
```

Comment:
Suggestion: for system debug, it's not very useful to define an ipc count. What would be more useful is to flood the system for e.g. 20ms or 200ms, so could we as an alternative use an ipc_flood_duration. This can then be used to write scripts to periodically flood for a fixed time, e.g 50ms every second, so that we can 'hear' the audio impact, if any

Reply:
@plbossart sure, I could do that. Do we want to retain both types though count and duration?